nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -275,11 +280,31 @@ BtNavigator::navigateToPose()
       geometry_msgs::msg::PoseStamped goal_pose;
       blackboard_->get("goal", goal_pose);
 
-      feedback_msg->distance_remaining = nav2_util::geometry_utils::euclidean_distance(
-        feedback_msg->current_pose.pose, goal_pose.pose);
+      // Get current path points
+      nav_msgs::msg::Path current_path;
+      blackboard_->get("path", current_path);
+
+      // Calculate distance on the path
+      double distance_remaining =
+        nav2_util::geometry_utils::calculate_path_length(current_path);
+
+      if (distance_remaining <= transform_tolerance_) {
+        estimated_navigation_time_remaining_ = rclcpp::Duration::from_seconds(0.0);
+      } else {
+        // Get current speed
+        current_linear_speed_ = odom_smoother_->getTwist().linear.x;
+
+        // Calculate estimated time taken to goal if speed is higher than 1mm/s
+        if (std::abs(current_linear_speed_) > 0.05) {
```

Comment:
What happens if the speed is lower?

Reply:
As the speed tends towards 0, the time remaining division tends to infinity, which is not what we want. That's why I added a threshold below which the time remaining won't update.

Reply:
Ah, it simply doesn't update of course.

But why don't we want infinity? I think it's the most correct value. Limiting to 5mm/s seems a bit arbitrary.

Reply:
Because divide by zeros are not reliable. I'd rather get a "I don't know" than a "NaN" or "Inf".

Reply:
Isn't NaN the definition of "I don't know" / "Cannot compute"

Reply:
I never want NaNs / Infs intentionally injected into my systems, as a personal thing. Error handling for them at all levels for all inputs is error prone and annoying

Reply:
I agree with Steve on this. Without this logic, the ETA, for example, might look like this 3, 2, 1, 0.5, 12, 345, 646746, 1e9 seconds... when you're almost at your goal, which doesn't really make sense.

@SteveMacenski does 5mm/s make sense? I was trying to tune this threshold but I wasn't sure what would be reasonable here.

Reply:
Ah, of course. You typically slow down on the end of the path. Having 3, 2, 1, 0.5, 12, 345, 646746 is terrible indeed!

I thought only of situations where you slow down the robot to zero halfway a path. For example on obstacles. In that case I thought `inf` to be a better estimation than 2 minutes.

Reply:
but it would be `3, 2, 1, 0.5, 0, 0, 0` if you set it to something sufficiently low that you really shouldn't see it except when actually stopped or very near to it (like 5 mm/s).